lib/PersonalityProvider.jsm

```diff
@@ -38,11 +66,67 @@ this.PersonalityProvider = class PersonalityProvider {
     this.scores = scores || {};
     this.interestConfig = this.scores.interestConfig;
     this.interestVector = this.scores.interestVector;
+    this.setupSyncAttachment(RECIPE_NAME);
+    this.setupSyncAttachment(MODELS_NAME);
+  }
+
+  setupSyncAttachment(collection) {
+    RemoteSettings(collection).on("sync", async event => {
+      const {
+        data: {created, updated, deleted},
+      } = event;
+
+      // Remove every removed attachment.
+      const toRemove = deleted.concat(updated.map(u => u.old));
+      await Promise.all(toRemove.map(record => this.deleteAttachment(record)));
+
+      // Download every new/updated attachment.
+      const toDownload = created.concat(updated.map(u => u.new));
+      await Promise.all(toDownload.map(record => this.downloadAttachment(record)));
+    });
+  }
+
+  async downloadAttachment(record) {
+    const {attachment: {location, filename}} = record;
+    const headers = new Headers();
+    headers.set("Accept-Encoding", "gzip");
+    // Still not quite working. Something up with promises.
+    // const resp = await fetch(`${baseAttachmentsURL}${location}`, {headers});
+    const resp = await fetch(`https://kinto.dev.mozaws.net/attachments/${location}`, {headers});
+    const buffer = await resp.arrayBuffer();
+    const bytes = new Uint8Array(buffer);
+    // const path = OS.Path.join(OS.Constants.Path.localProfileDir, PERSONALITY_PROVIDER_DIR_NAME, filename);
+    const path = OS.Path.join(OS.Constants.Path.localProfileDir, filename);
+    return OS.File.writeAtomic(path, bytes, {tmpPath: `${path}.tmp`});
+  }
+
+  deleteAttachment(record) {
+    const {attachment: {filename}} = record;
+    // const path = OS.Path.join(OS.Constants.Path.localProfileDir, PERSONALITY_PROVIDER_DIR_NAME, filename);
+    const path = OS.Path.join(OS.Constants.Path.localProfileDir, filename);
+    return OS.File.remove(path, {ignoreAbsent: true});
```

Comment:
`OS.File.removeDir(OS.Path.join(OS.Constants.Path.localProfileDir, PERSONALITY_PROVIDER_DIR_NAME), { ignoreAbsent: true });`

Reply:
I'm not sure if we need to remove the dir in general. Only if we're removing all the attachments do we need to remove the dir.

Are you thinking if we remove the dir but it has a file in it, does it just fail/catch or something?

Given that, the last attempt on the last file would work, fully cleaning up the dir?

Reply:
Oh, yeah that would be neat indeed :) 

TBH I just wanted to give you the snippet to delete the dir, I hadn't thought too much about the corner cases ;)

Reply:
Looks like this won't work as hoped, because from here: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/OSFile.jsm/OS.File_for_the_main_thread#OS.File.removeDir()

"Remove an existing directory and its contents."

Reply:
Here we go though: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/OSFile.jsm/OS.File_for_the_main_thread#OS.File.removeEmptyDir()